addons/mail/static/src/models/composer/composer.js

```diff
@@ -799,27 +687,16 @@ function factory(dependencies) {
             mainSuggestedRecords.length = Math.min(mainSuggestedRecords.length, limit);
             extraSuggestedRecords.length = Math.min(extraSuggestedRecords.length, limit - mainSuggestedRecords.length);
             this.update({
-                extraSuggestedRecords: replace(extraSuggestedRecords),
                 hasToScrollToActiveSuggestion: true,
+            });
+            this.suggestionList.update({
```

Comment:
The whole code of `_updateSuggestionList` should be done inside the suggestion list itself.

The main interaction between composer and suggestion list should be that the composer is updating the search term.